### PR TITLE
Fixed squared FAB on Android 4.4 (small screens)

### PR DESCRIPTION
**Pull request by amatsegor**
Also, added fabBackgroundColor attribute

### CODE REVIEW

**Comment by amatsegor** [comment]
What was the problem, actually
https://pp.vk.me/c837221/v837221724/2581a/p4Fla-LLcIg.jpg

**Comment by markormesher** [comment]
For the time being, could you move work for the background colour attribute to a new PR please? It's going to introduce a few breaking changes for existing users, so I'd rather keep it separate for now.

Also a few comments on the work to fix the squared FAB:
- There are still a lot of changes that are only line breaks or indentation, which make the diff harder to read (and break the style of the project). If you take a look at the diff and at existing files, you'll see the conventions that have been set in the project for indentation, XML formatting, etc.-
- Make sure you update documentation and comments when you update the code they apply to. For example, where you've changed `CardView` for `ImageButton`, the documentation still says `CardView`.
- Be aware of the implications of switching the `CardView` for an `ImageView` - make sure things like elevation and the background ripple effect stay consistent.

I apologise if any of this feel "picky" - I do appreciate your contribution! Feel free to make updates and comments in this PR; no need to delete and make a new one :smile:

**Comment by amatsegor** [comment]
Ok, will move it now

Sorry for some unchanged docs and code, I'll recheck and fix it now.

After switching from CardView, I've checked all states either on >A5.0 and <=A4.4, so it's got to work okay.

No problem at all, I agree with your feedback)

**Comment by amatsegor** [comment]
@markormesher done
Fixed button states on pre-Lollipop

File: app/build.gradle [hunk 1]
```diff
@@ -70,7 +70,7 @@ android {
 		debug {
 			// signing
 			//noinspection GroovyAssignabilityCheck
-			signingConfig signingConfigs.main
+//			signingConfig signingConfigs.main
```
Comment (markormesher) [inline app/build.gradle:73]: Why was this commented out?

File: app/src/main/res/layout/fab_activity.xml [hunk 2]
```diff
@@ -66,9 +66,9 @@
...
 	</ScrollView>
 
-	<uk.co.markormesher.android_fab.FloatingActionButton
-		android:id="@+id/fab"
-		android:layout_width="match_parent"
-		android:layout_height="match_parent"/>
+    <uk.co.markormesher.android_fab.FloatingActionButton
+        android:id="@+id/fab"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"/>
```
Comment (markormesher) [inline app/src/main/res/layout/fab_activity.xml:72]: Tabs/line breaks.

File: app/src/main/res/layout/fab_activity.xml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
-<RelativeLayout
-	xmlns:android="http://schemas.android.com/apk/res/android"
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+	xmlns:app="http://schemas.android.com/apk/res-auto"
```
Comment (markormesher) [inline app/src/main/res/layout/fab_activity.xml:3]: Tabs/line breaks.

File: fab/src/main/java/uk/co/markormesher/android_fab/FloatingActionButton.java [hunk 4]
```diff
@@ -147,7 +150,10 @@ private void resetIcon() {
 	 */
 	public void setBackgroundColour(int colour) {
 		savedBgColour = colour;
-		cardView.setCardBackgroundColor(colour);
+		final Drawable background = fab.getBackground();
+		if (background == null) return;
+
```
Comment (markormesher) [inline fab/src/main/java/uk/co/markormesher/android_fab/FloatingActionButton.java:155]: This needs to report a failure if the change was aborted here. The `background` variable is also unnecessary - it's accessed just once, so you can put `fab.get...` in the `if` condition.

File: fab/src/main/java/uk/co/markormesher/android_fab/FloatingActionButton.java [hunk 5]
```diff
@@ -340,7 +346,7 @@ public void rebuildSpeedDialMenu() {
 			speedDialMenuItems.add(view);
 
 			// set background colour
-			((CardView) view.findViewById(R.id.card)).setCardBackgroundColor(menuAdapter.getBackgroundColour(i));
+            setViewBackgroundTintColour(view.findViewById(R.id.card), menuAdapter.getBackgroundColour(i));
```
Comment (markormesher) [inline fab/src/main/java/uk/co/markormesher/android_fab/FloatingActionButton.java:349]: Tabs/line breaks.

File: fab/src/main/res/drawable-v21/fab_cover_circle.xml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ripple android:color="#fff" xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:drawable="@drawable/fab_round_shape"/>
+</ripple>
```
Comment (markormesher) [inline fab/src/main/res/drawable-v21/fab_cover_circle.xml:4]: Tabs/line breaks.

Comment (markormesher) [inline fab/src/main/res/drawable-v21/fab_cover_circle.xml:4]: Please use 6- or 8-digit colour codes to keep consistency with the rest of the project.

File: fab/src/main/res/drawable/fab_cover_circle.xml [hunk 7]
```diff
@@ -1,8 +1,13 @@
...
+        <shape android:shape="oval">
+            <solid android:color="#eee" />
+        </shape>
+    </item>
+    <item>
+        <shape android:shape="oval">
+            <solid android:color="#ffffffff" />
+        </shape>
+    </item>
+</selector>
```
Comment (markormesher) [inline fab/src/main/res/drawable/fab_cover_circle.xml:13]: Please use 6- or 8-digit colour codes to keep consistency with the rest of the project.

Comment (markormesher) [inline fab/src/main/res/drawable/fab_cover_circle.xml:13]: Tabs/line breaks.

File: fab/src/main/res/drawable/fab_round_shape.xml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<shape android:shape="oval" xmlns:android="http://schemas.android.com/apk/res/android">
+    <solid android:color="#fff"/>
+</shape>
```
Comment (markormesher) [inline fab/src/main/res/drawable/fab_round_shape.xml:4]: Please use 6- or 8-digit colour codes to keep consistency with the rest of the project.

Comment (markormesher) [inline fab/src/main/res/drawable/fab_round_shape.xml:4]: Tabs/line breaks.

File: fab/src/main/res/layout/floating_action_button.xml [hunk 9]
```diff
@@ -1,48 +1,52 @@
...
+            android:layout_height="@dimen/fab_icon_size"
+            android:layout_margin="@dimen/fab_icon_margin"
+            android:gravity="center"
+            android:orientation="horizontal" />
+
+    </FrameLayout>
 
-	</android.support.v7.widget.CardView>
 
 </RelativeLayout>
```
Comment (markormesher) [inline fab/src/main/res/layout/floating_action_button.xml:52]: Tabs/line breaks throughout.

File: fab/src/main/res/layout/speed_dial_icon.xml [hunk 10]
```diff
@@ -1,34 +1,39 @@
...
+            android:layout_width="@dimen/speed_dial_disc_icon_size"
+            android:layout_height="@dimen/speed_dial_disc_icon_size"
+            android:layout_margin="@dimen/speed_dial_disc_icon_margin"
+            android:elevation="2dp"
+            android:gravity="center"
+            android:orientation="horizontal" />
+
+    </FrameLayout>
 
 </LinearLayout>
```
Comment (markormesher) [inline fab/src/main/res/layout/speed_dial_icon.xml:39]: Tabs/line breaks throughout.

**Review by markormesher — CHANGES_REQUESTED** [review]
I've made some comments on areas that need to be adjusted to keep consistency with the rest of the project - mostly whitespace. You can check the "Files Changed" tab on GH when you make changes - it should only be highlighting your explicit code changes. Thanks again for your contributions so far :smiley:

**Comment by markormesher** [comment]
This seems to have gone cold and has been fixed anyway in v1.3.0. I'll close this, but thank you for your efforts!

**Pull request closed without merging**